Fix task.CooperativeTask in Python 3 due to change in StopIteration behaviour
--- a/src/twisted/internet/task.py
+++ b/src/twisted/internet/task.py
@@ -517,8 +517,11 @@
             result = next(self._iterator)
         except StopIteration:
             self._completeWith(TaskDone(), self._iterator)
-        except:
-            self._completeWith(TaskFailed(), Failure())
+        except BaseException as err:
+            if str(err) == "generator raised StopIteration":
+                self._completeWith(TaskDone(), self._iterator)
+            else:
+                self._completeWith(TaskFailed(), Failure())
         else:
             if isinstance(result, defer.Deferred):
                 self.pause()
